### PR TITLE
Added: `selector-max-universal` rule.

### PR DESCRIPTION
**Pull request by evilebottnawi**
> Which issue, if any, is this issue related to?

#2528

> Is there anything in the PR that needs further explanation?

No, it's self explanatory.

### CODE REVIEW

File: lib/rules/selector-max-universal/__tests__/index.js [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+  reject: [ {
+    code: "* * * {}",
+    description: "compound selector: greater than max universal selectors",
+    message: messages.expected("* * *", 2),
+    line: 1,
+    column: 1,
+  }, {
+    code: "*, \n* * * {}",
+    description: "multiple selectors: greater than max classes",
```
Comment (hudochenkov) [inline lib/rules/selector-max-universal/__tests__/index.js]: "multiple selectors: greater than max universal selectors"

File: lib/rules/selector-max-universal/README.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+.foo * {}
+```
+
+```css
+*.foo * {}
+```
+
+```css
+/* each selector in a selector list is evaluated separately */
+*.foo *,
```
Comment (jeddy3) [inline lib/rules/selector-max-universal/README.md]: Please change to:

```css
*.foo,
*.bar * {}
```

That that only three (one more than the max) universal selectors are used.

File: lib/rules/selector-max-universal/README.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+*.bar * {}
+```
+
+```css
+/* `*` is inside `:not()`, so it is evaluated separately */
```
Comment (jeddy3) [inline lib/rules/selector-max-universal/README.md]: Please change to:

```css
* > * .foo:not(*) {}
```

**Comment by jeddy3** [comment]
@hudochenkov I've made the changes if you'd like to look them over?

**Comment by hudochenkov** [comment]
@jeddy3 looks good. Thanks!

**Comment by jeddy3** [comment]
> -   Added: `selector-max-universal` rule ([#2653](https://github.com/stylelint/stylelint/pull/2653)).

**Pull request merged**